metrics_utility/automation_controller_billing/helpers.py

```diff
@@ -9,7 +9,11 @@
 from dateutil import parser
 from dateutil.relativedelta import relativedelta
 
-from metrics_utility.exceptions import UnparsableParameter
+from metrics_utility.exceptions import DateFormatError, UnparsableParameter
+
+
+DATE_REGEX_PATTERN = r'^\d{4}-\d{2}(-\d{2})?$'
```

Comment:
I wonder if this could be too restrictive?

Places using `dateutil.parser.parse` would previously also acccept dates like "4/5/2063", "15.5.2011", and such. 

Would it make sense to only use this for the month check? Then again I wonder if anyone is using these other formats, maybe it IS safe to require iso dates.

Reply:
No, I think you are right.  We should use parser.parse.  As for the month check, we have a current check happening [here](https://github.com/ansible/metrics-utility/pull/128/files/1a7be4180a36ccf06743818c0a46581ae3e6316b).  Do you think that will suffice?

Reply:
So I ended up just removing that regex and opting for a new function to help validate expected date/time values.